modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -198,8 +198,40 @@ sda.enkf <- function(settings, obs.mean, obs.cov, Q = NULL, restart=F,
                         ensemble.id=ensemble.id)
       
     }else{
-      restart.arg <- NULL
```

Comment:
It's hard for me to ask for a change before running the code but I need to clarify something here. In general in the re-factored sda the first run (t==1), writing the config is done by general `write.config.model` function whereas in the t>1 writing the config is directed to `write.restart.config`. So here the in L201 the `restart.arg` variable is responsible for carrying the required inputs for `write.restart.config` and has nothing to do with restart argument of whole sda . That restart is setup in line 144 instead. Now based on  what I just explained see if your code here needs to be moved to L145 or it's in the right place.

Reply:
So you think restart.arg and input.split for restart == TRUE should be taken care of outside of the loop over time because restart.arg is only for write.restart.config?

Reply:
Now, reading your code I think if you just move Line 219-227 to the scope of `if` in line 144. It should work and do  what you expect and you won't need the rest (splitting and stuff).

Reply:
So I think what the block between 201 and 233 is for  is for when spin up has been run but no forecast so I think I'd still need the split inputs right and to leave in the time loop?

Reply:
That's true you need them, but I think if you move that to L144 then it would go with the logic of the code before that and the split will happen in line 178. Because  if you think about it, when you `restart==TRUE` then it loads your `sda.output.Rdata` and then `t` would be greater than 1 and then  it would do the split and so on.